hmcts/pre-api
# Fix: `PUT /case/{id}` Constraints Surrounding Case State and ClosedAt

### JIRA ticket(s)
- S28-3088


### Change description
- Removed db constraints for `state` and `closedAt` columns and moved them to the API
- Case `state` column now `NOT NULL` and defaults to `OPEN`
- Type of `closed_at` column updated `date` -> `timestamptz` (updated API for this change)
- When state is not specified in request (state=null) state is set to OPEN (db default)
- Added constraints to `PUT /case/{id}` endpoint
    - When state = OPEN, closedAt must be null
    - When state != OPEN, closedAt must not be null
    - When state = CLOSED, closedAt must not be in the future

**If your db has already been updated with the changes from the `S28-3088/disassociation2` PR then rollback the migration and rerun with the updated one**




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Test: ensure change is non-breaking - sending requests without values for `state` and `closed_at` should not result in an error (functionality should be preserved)
- Test the happy and unhappy paths for each of the constraints listed in the description
